src/common/Common.ts

```diff
@@ -75,3 +75,43 @@ export class CommandLineUtils {
         }
     }
 }
+
+// tslint:disable-next-line: max-classes-per-file
+export class Version {
+    public readonly major: number;
+    public readonly minor: number;
+    public readonly patch: number;
+
+    constructor(from: string) {
```

Comment:
I would prefer to see a slightly different pattern here.

1. Move the "fromString" parsing functionality of the constructor into a factory method. This gives us the opportunity to parse the version string for validity, and return `null` if it's an invalid string.
2. Following from 1, the "sanity checks" in `sameOrNewer` should be retired in favor of ensuring that a `Version` object implicitly represents a proper version.

Reply:
Sorry, the part I forgot to say out loud 😀: in our factory method, we should parse for a valid version string as the sanity check, e.g. `/\d+([-\.]\d+)*/`, and return `null` if the input doesn't meet that requirement. Otherwise, the code below parses it out and sends it to a major/minor/patch parameterized constructor.

Reply:
👍

Reply:
@khawkins for #3, how about we just throw have the factory method throw an exception if the input is invalid (instead of returning null)? Would that be ok too or you prefer null? Both or fine to me.

Reply:
I'm okay with either. I recommended `null` because I wasn't sure if throwing an error would be too heavy-handed for a data object. But honestly, whatever fits our code flow best, between the two, is fine by me.